File: DemoCollectOC/DemoCollectOC/MainViewController.m
//
//  MainViewController.m
//  DemoCollectOC
//
//  Created by Shawn.Du on 16/3/31.
//  Copyright © 2016年 Shawn.Du. All rights reserved.
//

#import "MainViewController.h"

@interface MainViewController ()<UITableViewDelegate, UITableViewDataSource>

@end

@implementation MainViewController {
    UITableView *_tableView;
    NSArray *_classNames;
}

#pragma mark - life cycle
- (void)viewDidLoad {
    [super viewDidLoad];
    _classNames = @[@"AppListViewController", @"GifDisplayViewController"];
    _tableView = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStylePlain];
    _tableView.delegate = self;
    _tableView.dataSource = self;
    [self.view addSubview:_tableView];
}

#pragma mark - tableView datasouce

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _classNames.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *identifier = @"MainTableIdentifier";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identifier];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:identifier];
    }
    cell.textLabel.text = _classNames[indexPath.row];
    return cell;
}


#pragma mark tableView delegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    Class cls = NSClassFromString(_classNames[indexPath.row]);
    if (cls) {
        UIViewController *ctrl = cls.new;
        ctrl.title = _classNames[indexPath.row];
        [self.navigationController pushViewController:ctrl animated:YES];
    }
    [_tableView deselectRowAtIndexPath:indexPath animated:YES];
    
}

@end

File: DemoCollect/DemoCollect/MainViewController.swift
//
//  MainViewController.swift
//  DemoCollect
//
//  Created by dulingkang on 16/3/15.
//  Copyright © 2016年 com.shawn. All rights reserved.
//

import UIKit


class MainViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {

    var classes: [String] {
        get{
            return ["AnchorPoint",
                "NaviFrame",
                "StringStatic",
                "PriceChoose",
                "PathAnimation"
            ]
        }
    }
    
    //MARK: - life cycle
    override func viewDidLoad() {
        configViews()
    }
    
    //MARK: - tableView
    //MARK: dataSource
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return classes.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let identifier = "SSMainCell"
        var cell = tableView.dequeueReusableCell(withIdentifier: identifier)
        if (cell == nil) {
            cell = UITableViewCell(style: .default, reuseIdentifier: identifier)
        }
        cell?.textLabel?.text = classes[indexPath.row]
        return cell!
    }
    
    //MARK: delegte
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        var className = classes[indexPath.row]
        className = "DemoCollect" + "." + className
        let currentClass = NSClassFromString(className) as! UIViewController.Type
        let destVC = currentClass.init()
        self.navigationController?.pushViewController(destVC, animated: true)
    }
    
    //MARK: - private method
    fileprivate func configViews() {
        let tableView = UITableView(frame: self.view.bounds, style: .plain)
        tableView.delegate = self
        tableView.dataSource = self
        view.addSubview(tableView)
    }
}
